server/util/file.py

```diff
@@ -74,6 +74,23 @@ def retrieve_file(storage_path: str) -> BinaryIO:
         return open(storage_path, "rb")
 
 
+def retrieve_file_streaming(storage_path: str, working_directory: str) -> BinaryIO:
```

Comment:
My only nit is that `retrieve_file_streaming` returns the same streamable interface as `retrieve_file`, so it feels like the naming is a bit confusing. I think the key difference seems to be that one buffers the file in memory, while one saves it to disk. So maybe we could call them `retrieve_file_to_buffer` and `retrieve_file_to_temp_file` or something like that?

And it would probably be good to leave a comment ensuring that the caller of this function knows that they are responsible for cleaning up `working_directory` after they are done.